packages/mermaid/src/diagrams/git/gitGraphAst.js

```diff
@@ -274,21 +275,21 @@ export const cherryPick = function (sourceId, targetId, tag) {
     };
     throw error;
   }
-
   let sourceCommit = commits[sourceId];
   let sourceCommitBranch = sourceCommit.branch;
   if (sourceCommit.type === commitType.MERGE) {
-    let error = new Error(
-      'Incorrect usage of "cherryPick". Source commit should not be a merge commit'
-    );
-    error.hash = {
-      text: 'cherryPick ' + sourceId + ' ' + targetId,
-      token: 'cherryPick ' + sourceId + ' ' + targetId,
-      line: '1',
-      loc: { first_line: 1, last_line: 1, first_column: 1, last_column: 1 },
-      expected: ['cherry-pick abc'],
-    };
-    throw error;
+    if (!parentCommitId) {
+      let error = new Error(
+        'Incorrect usage of cherry-pick: If the source commit is a merge commit, an immediate parent commit must be specified.'
+      );
+      throw error;
+    }
+    if (!(Array.isArray(sourceCommit.parents) && sourceCommit.parents.includes(parentCommitId))) {
+      let error = new Error(
+        'Invalid operation: The specified parent commit is not an immediate parent of the cherry-picked commit.'
+      );
+      throw error;
+    }
```

Comment:
One potential change (although we could do this in another PR, since I know this PR has been open for a while). Should we move this check outside of the `if (sourceCommit.type === commitType.MERGE) {` condition?

```suggestion
  }
  if (parentCommitId && !(Array.isArray(sourceCommit.parents) && sourceCommit.parents.includes(parentCommitId))) {
    let error = new Error(
      'Invalid operation: The specified parent commit is not an immediate parent of the cherry-picked commit.'
    );
    throw error;
```

Currently, if we use a bad `parent: "xxx"` field for a cherry-pick that **doesn't target a MERGE**, there's no error, e.g. the following diagram renders correctly without any errors:

````markdown
```mermaid
gitGraph
    commit id: "ZERO"
    branch feature
    branch release
    checkout feature
    commit id: "A"
    commit id: "Not a merge commit"
    checkout main
    merge feature id: "M"
    checkout release
    cherry-pick id: "Not a merge commit" parent: "this-parent-does-not-exist"
```
````

Reply:
Hello @aloisklink , Thanks for this suggestion , but TBH, I didn't understand the reason for   moving out this `if (sourceCommit.type === commitType.MERGE) ` condition. My take on this was that, first we should check the commit type, because we will not be having `parentCommitId`, for a commit which is not a merge commit.  Could you please explain where I am lagging in getting your thoughts behind this approach, and if you feel this should include in this PR, I will be very happy to do that, otherwise I will immediately raise a new PR solving both issues.
Thanks.

Reply:
Hello @aloisklink , Now I understood your point clearly , and feeling that this thing needs to be solved in this PR only, thanks for pointing this out. Fixing and pushing the commit, as soon as possible. 
Thanks

Reply:
Fixed by https://github.com/mermaid-js/mermaid/pull/4944/commits/31a1de156685539008ffe666ad1ce1ca3435bb71 and https://github.com/mermaid-js/mermaid/pull/4944/commits/d22ee8d1d55eb278d61272d69c7a467ed080b1c9! Thanks @RounakJoshi09 and @sidharthv96